mii/deployment.py

```diff
@@ -98,6 +100,23 @@ def deploy(task,
     elif model_path is None and deployment_type == DeploymentType.AML:
         model_path = "model"
 
+    # add fields for replica deployment
+    replica_pool = _allocate_processes(mii_config.hostfile,
+                                       mii_config.tensor_parallel,
+                                       mii_config.replica_num)
+    replica_deployment = []
+    for i, (host, gpu_indices) in enumerate(replica_pool):
+        grpc_ports = []
+        # Reserver port for a LB proxy when replication is enabled
+        port_offset = 1 if mii_config.replica_num > 1 else 0
+        for j in range(mii_config.tensor_parallel):
+            grpc_ports.append(mii_config.port_number + i * mii_config.tensor_parallel +
+                              j + port_offset)
```

Comment:
This could be simplified:
```python
port_offset = 1 if mii_config.replica_num > 1 else 0
base_port = mii_config.port_number + i * mii_config.tensor_parallel + port_offset
grpc_ports = range(base_port, base_port+mii_config.tensor_parallel)